src/ray/raylet/scheduling/local_resource_manager.cc

```diff
@@ -501,11 +501,20 @@ ray::gcs::NodeResourceInfoAccessor::ResourceMap LocalResourceManager::GetResourc
 }
 
 void LocalResourceManager::OnResourceChanged() {
+  if (IsLocalNodeDraining() && IsLocalNodeIdle()) {
```

Comment:
Can we add unit tests for this case?

1. drain & none idle -> no terminate
2. and then it becomes idle -> terminate

Reply:
maybe also good to test basic cases